Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2283 -- Search box in Policy Repo sidebar

Resolves [EREGCSC-2283](https://jiraent.cms.gov/browse/EREGCSC-2283)

**Description:**

Add the search box and integrate the search results to existing policy repository page.

**This pull request changes:**

- Refactors logic that watches `$route.query` and updates the page.  Removes lots of complexity.
- Adds `SearchInput.vue` component to the left sidebar of the Policy Repository page and uses it to add keyword search

**Steps to manually verify this change:**

1. Visit the [experimental deployment](https://w5w458srf0.execute-api.us-east-1.amazonaws.com/dev986)
2. On the policy repository page, test the search input functionality
3. Make sure you can add and remove subjects with and without a search query, etc.
4. Make sure all tests pass
